### PR TITLE
Initial commit of new job APIs

### PR DESCRIPTION
**Pull request by vamsi-amazon**
### Description
* Contains Rest, Transport Layer Files for new job APIs.
* Currently we are making the changes in spark connector module, we may move to a new module once the design becomes concrete.
* Documentation will be added in individual API PRs.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: spark/src/main/java/org/opensearch/sql/spark/transport/TransportCreateJobRequestAction.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.sql.spark.transport.model.CreateJobActionRequest;
+import org.opensearch.sql.spark.transport.model.CreateJobActionResponse;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+
+public class TransportCreateJobRequestAction
+    extends HandledTransportAction<CreateJobActionRequest, CreateJobActionResponse> {
+
+  public static final String NAME = "cluster:admin/opensearch/ql/jobs/write";
```
Comment (penghuo) [inline spark/src/main/java/org/opensearch/sql/spark/transport/TransportCreateJobRequestAction.java:23]: align with datasource creation? `cluster:admin/opensearch/ql/jobs/create`

File: spark/src/main/java/org/opensearch/sql/spark/transport/TransportGetJobRequestAction.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.sql.spark.transport.model.GetJobActionRequest;
+import org.opensearch.sql.spark.transport.model.GetJobActionResponse;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+
+public class TransportGetJobRequestAction
+    extends HandledTransportAction<GetJobActionRequest, GetJobActionResponse> {
+
+  public static final String NAME = "cluster:admin/opensearch/ql/jobs/read";
```
Comment (penghuo) [inline spark/src/main/java/org/opensearch/sql/spark/transport/TransportGetJobRequestAction.java:23]: user has permission on `cluster:admin/opensearch/ql/jobs/read` can access
1. `_plugins/_query/jobs` list all the job status.
2. `_plugins/_query/jobs/{jobId}` fetch specify job status.

File: spark/src/main/java/org/opensearch/sql/spark/transport/TransportGetQueryResultRequestAction.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import org.opensearch.sql.spark.transport.model.GetJobQueryResultActionRequest;
+import org.opensearch.sql.spark.transport.model.GetJobQueryResultActionResponse;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+
+public class TransportGetQueryResultRequestAction
+    extends HandledTransportAction<
+        GetJobQueryResultActionRequest, GetJobQueryResultActionResponse> {
+
+  public static final String NAME = "cluster:admin/opensearch/ql/jobs/result";
```
Comment (penghuo) [inline spark/src/main/java/org/opensearch/sql/spark/transport/TransportGetQueryResultRequestAction.java:24]: what is the different of result and status? result API include status, right?

File: spark/src/main/java/org/opensearch/sql/spark/transport/TransportGetJobRequestAction.java (hunk 2 above)
Comment (vamsi-amazon) [reply spark/src/main/java/org/opensearch/sql/spark/transport/TransportGetJobRequestAction.java:23]: yes.

File: spark/src/main/java/org/opensearch/sql/spark/transport/TransportGetQueryResultRequestAction.java (hunk 3 above)
Comment (vamsi-amazon) [reply spark/src/main/java/org/opensearch/sql/spark/transport/TransportGetQueryResultRequestAction.java:24]: yes. Something like below.
```
{
      "status" : "IN_PROGRESS"
}
```
```
{
      "status" : "COMPLETED",
      "schema" : {},
      "result" : {}
}
```

**Review by penghuo — APPROVED** [review]

**Pull request merged**